src/OaiPmh/Harvester.php

```diff
@@ -340,10 +336,20 @@ protected function getRecordsByToken($token)
      */
     protected function storeDateSettings($settings)
     {
+        // Get identity information, including current OAI host time.
+        $response = $this->sendRequest('Identify');
+        $servertime = (string)$response->responseDate;
```

Comment:
Public note to self... I'm actually thinking that this call should be moved inside of the launch() method. storeDateSettings(), and previously loadgranularity() (where this call originally lived), get called from the constructor. This feels like something that should be moved outside of the constructor, especially as an external call is needed. Having it inside launch() may also make managing returned values a bit easier, since that's the scope where they will most likely be needed.

Reply:
Agreed, it's fairly bad form for the constructor to have side effects. It might also make sense to set up smart getters -- i.e. rather than fetching the Identify response in advance, we have a getIdentifyResponse() method that returns a property if defined or pulls from the server if undefined. Then anything that needs data from there can call getIdentifyResponse... and then if the code gets reorganized or refactored, it will still be smart enough to pull what it needs on demand and not do redundant work.

Reply:
Yes, I think that would improve the DX quite a bit. https://github.com/vufind-org/vufindharvest/pull/8/commits/569569a47dd72eae273aec256a15cae07c093a9e adds a getIdentifyResponse() method such that we can get related properties (over and over) with nice chained calls, like:
```
$this->getIdentifyResponse()->granularity;
$this->getIdentifyResponse()->responseDate;
```